src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetAutoMappedLocationsTable.tsx

```diff
@@ -1,61 +1,176 @@
+        location =>
+          location.candidate.label
+            .toLowerCase()
+            .includes(searchTerm.toLowerCase()) ||
+          location.mapping.source.label
+            .toLowerCase()
+            .includes(searchTerm.toLowerCase()) ||
+          location.candidate.code
+            ?.toLowerCase()
+            .includes(searchTerm.toLowerCase()) ||
+          location.mapping.source.code
+            ?.toLowerCase()
+            .includes(searchTerm.toLowerCase()),
+      )
+    : locations;
+  const filteredLocationChunks = chunk(filteredLocations, locationsPerPage);
+  const totalFilteredLocations = filteredLocations.length;
+  const totalPages = filteredLocationChunks.length;
+
   return (
-    <table className={styles.table} data-testid="auto-mapped">
-      <caption className="govuk-visually-hidden">
-        Table showing all auto mapped locations
-      </caption>
-      <thead>
-        <tr>
-          <th className="govuk-!-width-one-third">Current data set</th>
-          <th className="govuk-!-width-one-third">New data set</th>
-          <th>Type</th>
-          <th>Actions</th>
-        </tr>
-      </thead>
-      <tbody>
-        {/* TODO EES-5325 add search and pagination, just showing first ten for now. */}
-        {locations.slice(0, 10).map(({ candidate, mapping }) => {
-          return (
-            <tr key={`mapping-${mapping.source.code}`}>
-              <td>
-                {mapping.source.label}
-                <br />
-                <span className={styles.code}>{mapping.source.code}</span>
-              </td>
-              <td>
-                {candidate.label}
-                <br />
-                <span className={styles.code}>{candidate.code}</span>
-              </td>
-              <td>
-                <Tag colour="grey">Minor</Tag>
-              </td>
-              <td>
-                <ButtonText>
-                  Edit
-                  <VisuallyHidden>
-                    {' '}
-                    mapping for {mapping.source.label}
-                  </VisuallyHidden>
-                </ButtonText>
-              </td>
+    <>
+      <div className="govuk-grid-row">
+        <div className="govuk-grid-column-one-half">
+          <form
+            id={formId}
+            className="govuk-!-margin-bottom-2"
+            onSubmit={e => {
+              e.preventDefault();
+            }}
+          >
+            <FormSearchBar
+              id={`${formId}-search`}
+              label="Search auto mapped locations"
+              labelSize="s"
+              min={0}
+              name="search"
+              value={searchTerm}
+              onChange={term => {
+                setSearchTerm(term);
+                setCurrentPage(1);
+              }}
+            />
+          </form>
+          <VisuallyHidden>
+            <p aria-atomic aria-live="polite">
+              {`${totalFilteredLocations} results, showing page ${currentPage} of ${totalPages}`}
```

Comment:
I was just wondering  - does pluralisation affect screen readers much? I assume it's not too much of a problem given its just grammar and it will read as '1 results'

Reply:
Good spot, I've changed it to have the correct pluralisation now.